### PR TITLE
PublicDashboards: Time range conditionally shown

### PR DESCRIPTION
**Pull request by juanicabanas**
**What is this feature?**

Public dashboards: time range picker is conditionally shown, based on public dashboard flag configuration

**Why do we need this feature?**

It shows or hides the time range picker in the public dashboard

**Which issue(s) does this PR fix?**:

Fixes [#440](https://github.com/grafana/grafana-enterprise/issues/4427)

**Special notes for your reviewer**:

https://user-images.githubusercontent.com/11707172/207964695-8817c848-f9a6-404f-b803-5e64f9d961b5.mov

### CODE REVIEW

**Review by jalevin — APPROVED** [review]
LGTM. 1 nitpick, but I think its sound.

File: public/app/features/dashboard/containers/DashboardPage.tsx [hunk 1]
```diff
@@ -352,7 +352,8 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
     const inspectPanel = this.getInspectPanel();
     const showSubMenu = !editPanel && !kioskMode && !this.props.queryParams.editview;
 
-    const toolbar = kioskMode !== KioskMode.Full && !queryParams.editview && (
+    const toolbar = ((kioskMode !== KioskMode.Full && !queryParams.editview) ||
```
[the remark this answers is not in the document]
Comment (juanicabanas) [reply public/app/features/dashboard/containers/DashboardPage.tsx:355]: just added!

File: public/app/features/dashboard/components/DashNav/DashNav.tsx [hunk 2]
```diff
@@ -321,6 +322,10 @@ export const DashNav = React.memo<Props>((props) => {
       return [renderPlaylistControls(), renderTimeControls()];
     }
 
+    if (props.isPublic && !!props.dashboard.meta.publicDashboardTimeSelectionEnabled) {
+      return [renderTimeControls()];
+    }
```
Comment (torkelo) [inline public/app/features/dashboard/components/DashNav/DashNav.tsx:327]: @juanicabanas for example both these conditionals can be removed, and you can just use the dashboard model's hide hideTimePicker option instead `dashboard.timepicker.hidden`  (set in backend to false)

Comment (torkelo) [reply public/app/features/dashboard/components/DashNav/DashNav.tsx:327]: @juanicabanas and this condition /option is already checked inside renderTimeControls

**Pull request merged**